sunpy/timeseries/metadata.py

```diff
@@ -0,0 +1,366 @@
+                self.metadata.append(meta)
+            elif isinstance(meta, list):
+                # Given a complex metadata list (of tuples)
+                self.metadata = meta.copy()
+            
+        
+
+    def append(self, timerange, columns, metadata, **kwargs):
+        """
+        Add the given metadata MetaDict into the metadata list as a tuple.
+        Will add the new entry so the list is in chronological order for the
+        TimeRange.start datetime values.
+        
+        Parameters
+        ----------
+        timerange : `~sunpy.time.TimeRange`
+            The string (parsed using the `~sunpy.time.parse_time`) or datetime
+            that you need metadata for.
+        
+        columns : `str`
+            A string that can be used to narrow results to specific columns.
+
+        metadata : `~sunpy.util.metadata.MetaDict`
+            A string that can be used to narrow results to specific columns.
+        """
+        # Check the types are correct.
+        pos = len(self.metadata)
+        if isinstance(timerange, TimeRange) and isinstance(metadata, MetaDict):
+            for i in range(0, len(self.metadata)):
+                if timerange.start < self.metadata[i][0].start:
+                    pos = i - 1
+        else:
+            raise ValueError(
+                'Incorrect datatime or data for append to TimeSeriesMetaData.')
+        
+        # Prepare tuple to append.
+        new_metadata = (timerange, columns, metadata)
+        
+        # Check this isn't a duplicate entry (same TR and comnames)
+        duplicate = False
+        if pos < len(self.metadata):
+            old_metadata = self.metadata[pos]
+            if (new_metadata[0].start == old_metadata[0].start) and (new_metadata[0].end == old_metadata[0].end) and (new_metadata[1] == old_metadata[1]):
+                duplicate = True
+                
+        # Insert into the given position
+        if not duplicate:
+            self.metadata.insert(pos, new_metadata)
+            
+    def find(self, **kwargs):
+        """
+        Find all metadata matching the given datetime and optionally column name.
+        
+        Parameters
+        ----------
+        datetime : `str` or `~datetime.datetime` optional
+            The string (parsed using the `~sunpy.time.parse_time`) or datetime
+            that you need metadata for.
+        
+        colname : `str` optional
+            A string that can be used to narrow results to specific columns.
+
+        indexes : `bool` optional
+            If True then return a list of indexes, not of MetaDict items.
+    
+        Returns
+        -------
+        list : `list`
+            A list of MetaDict objects that contain all matching metadata.
+        """
+        # Parameters
+        datetime = kwargs.get('datetime', None)
+        colname = kwargs.get('colname', None)        
+        indexes = kwargs.get('indexes', False)
+        dt = datetime

```

Comment:
why rename `datetime` to `dt` here? (also using `datetime` as a variable name is probably not a great idea as it might conflict with the module.